Cookbook/Proposals/MocksAndStubsNextToImplementation.md

```diff
@@ -0,0 +1,53 @@
+# Put Mocks, Stubs Next to Implementation
+
+* Author(s): Sergey Shulga
+* Review Manager: TBD
+
+## Introduction
+
+Put implementationf of mocks next to the implementation of the protocol
+
+## Motivation
+
+
+- Improves Desccoverability of mocks
+- Prevents mocks duplication (Because we usually define mocks in the `FrameworkTests` target which means they can only be used there)
+- Easy to change mock implementation once public API changes
```

Comment:
By this I mean that we would not need to wait for Tests to fail to compile when we change interface of the protocol

Reply:
How does it become easier to change the mock implementation?

Reply:
I mean that when we build main app the mocks won't compile as well, so compiler would force us to modify mocks as well, instead of doing it later when working on tests

Reply:
I understood that according to proposal and comments on it, these files will be excluded when building the app. So, I am not sure if this will make any difference.